index.js

```diff
@@ -6,8 +6,8 @@ var util = require('util');
  * Regexps
  */
 
-var DATE_TYPE_RE = /^\d{4}\-\d{2}\-\d{2}$/;
-var DATETIME_TYPE_RE = /^\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}$/;
+var DATE_TYPE_RE = /^(?:[1-9]\d{3}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1\d|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[1-9]\d(?:0[48]|[2468][048]|[13579][26])|(?:[2468][048]|[13579][26])00)-02-29)$/;
```

Comment:
where dose these regexps come from? It is a little complex, can you provide a tiny benchmark?